indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -56,13 +64,17 @@ public class SupervisorManager
   // SupervisorTaskAutoScaler could be null
   private final ConcurrentHashMap<String, SupervisorTaskAutoScaler> autoscalers = new ConcurrentHashMap<>();
   private final Object lock = new Object();
+  private final ListeningExecutorService shutdownExec;
 
   private volatile boolean started = false;
 
   @Inject
   public SupervisorManager(MetadataSupervisorManager metadataSupervisorManager)
   {
     this.metadataSupervisorManager = metadataSupervisorManager;
+    this.shutdownExec = MoreExecutors.listeningDecorator(
+        Execs.multiThreaded(25, "supervisor-manager-shutdown-%d")
```

Comment:
the issue i was running into with this strategy is that part of the stop logic is shutting down the scheduledExec executor, and I couldn't really think of a great way to avoid this chicken-and-egg problem.

Reply:
You could perhaps work around that problem by doing something like this:
- `stopAsync` sets the supervisor state to STOPPING
- `stopAsync` then submits a `stop()` runnable to the `scheduledExec`
- `buildRunTask` method should check and submit the `RunNotice` only if the state of the supervisor is not STOPPING
- `stop()` can call `scheduledExec.shutdown()` instead of `scheduledExec.shutdownNow()`

Let me know what you think.

Another alternative is to simply create a `shutdownExec` inside `stopAsync`.
Difference from the current approach would be that the `SupervisorManager` doesn't need to handle the lifecycle of the shutdown executor.

Reply:
i think this makes sense, will update